src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs

```diff
@@ -58,8 +60,17 @@ impl InputFormatCSV {
                     let err_msg = format_column_error(schema, c, col_data, &e.message());
                     return Err(csv_error(&err_msg, path, row_index));
                 };
-                if reader.must_eof().is_err() {
-                    let err_msg = format_column_error(schema, c, col_data, "bad field end");
+                let mut next = [0u8; 1];
+                let readn = reader.read(&mut next[..])?;
+                if readn > 0 {
+                    let remaining = col_data.len() - reader.pos + 1;
+                    let err_msg = format!(
+                        "bad field end: next={}, len={}.",
```

Comment:
At first glance, I can't understand what `next` and `len` mean. How about providing more context?

The trailing `.` can be removed.

Reply:
refined